libavcodec/videotoolboxenc.c

```diff
@@ -2065,63 +2069,16 @@ static int get_cv_pixel_info(
         }
     }
 
-    switch (av_format) {
-    case AV_PIX_FMT_NV12:
-        *plane_count = 2;
-
-        widths [0] = avctx->width;
-        heights[0] = avctx->height;
-        strides[0] = frame ? frame->linesize[0] : avctx->width;
-
-        widths [1] = (avctx->width  + 1) / 2;
-        heights[1] = (avctx->height + 1) / 2;
-        strides[1] = frame ? frame->linesize[1] : (avctx->width + 1) & -2;
-        break;
-
-    case AV_PIX_FMT_YUV420P:
-        *plane_count = 3;
-
-        widths [0] = avctx->width;
-        heights[0] = avctx->height;
-        strides[0] = frame ? frame->linesize[0] : avctx->width;
-
-        widths [1] = (avctx->width  + 1) / 2;
-        heights[1] = (avctx->height + 1) / 2;
-        strides[1] = frame ? frame->linesize[1] : (avctx->width + 1) / 2;
-
-        widths [2] = (avctx->width  + 1) / 2;
-        heights[2] = (avctx->height + 1) / 2;
-        strides[2] = frame ? frame->linesize[2] : (avctx->width + 1) / 2;
-        break;
+    *plane_count = av_pix_fmt_count_planes(avctx->pix_fmt);
 
-    case AV_PIX_FMT_BGRA:
-        *plane_count = 1;
-
-        widths [0] = avctx->width;
-        heights[0] = avctx->height;
-        strides[0] = frame ? frame->linesize[0] : avctx->width * 4;
-        break;
-
-    case AV_PIX_FMT_P010LE:
-        *plane_count = 2;
-        widths[0] = avctx->width;
-        heights[0] = avctx->height;
-        strides[0] = frame ? frame->linesize[0] : (avctx->width * 2 + 63) & -64;
-
-        widths[1] = (avctx->width + 1) / 2;
-        heights[1] = (avctx->height + 1) / 2;
-        strides[1] = frame ? frame->linesize[1] : ((avctx->width + 1) / 2 + 63) & -64;
-        break;
-
-    default:
-        av_log(
-               avctx,
-               AV_LOG_ERROR,
-               "Could not get frame format info for color %d range %d.\n",
-               av_format,
-               av_color_range);
-
-        return AVERROR(EINVAL);
+    for (i = 0; i < desc->nb_components; i++) {
+        int p = desc->comp[i].plane;
+        int chroma = (p != 0) && (!(desc->flags & AV_PIX_FMT_FLAG_ALPHA) || p + 1 != *plane_count);
```

Comment:
This could be broken out into more variables to make the logic clearer. It took me a while to parse this.